src/units/abilities.cpp

```diff
@@ -1168,18 +1168,24 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 	}
 }
 
-//Functions used for change damage_type list with damage
-static std::optional<std::string> select_damage_type(const unit_ability_list& abil_list, const std::string& type)
+//select damage type who must be returned for each attribute used in damage_type()
+std::optional<std::string> attack_type::select_damage_type(const unit_ability_list& abil_list, const std::string& type, bool is_alternative, unit_ability_list resistance_abilities) const
```

Comment:
With `is_alternative` the `type` argument seems obsolete.

Reply:
removed